src/Resources/contao/controllers/BackendMain.php

```diff
@@ -59,13 +61,7 @@ public function __construct()
 		// Password change required
 		if ($this->User->pwChange)
 		{
-			$objSession = $this->Database->prepare("SELECT su FROM tl_session WHERE hash=?")
-										 ->execute($this->getSessionHash('BE_USER_AUTH'));
-
-			if (!$objSession->su)
-			{
-				$this->redirect('contao/password.php');
-			}
+			$this->redirect('contao/password.php');
```

Comment:
We have to check for the admin role here, so the redirect is not executed if an admin impersonates another user.